### PR TITLE
General non-version-related documentation update to publish in 2.1.4

### PR DESCRIPTION
**Pull request by potiuk**
Cherry-picked some non-version related changes to doc - to publish them
still for current stable (2.1.4) before we release the 2.2 (that might
still take a few weeks).

The cherry-picks are pushed to v2-1-test (just in case we need to
release 2.1.5):

* Explain scheduler fine-tuning better (#18356) …
* Add PGBouncer recommendation in "setup-database' doc. (#18399) …
* Doc: Fix broken asc download link (#18350) …
* add sla_miss_callback section to the documentation (#18305) …

### CODE REVIEW

**Comment by potiuk** [comment]
HEy @kaxil - I cherry-picked some non-version related changes to doc - to publish them still for current stable (2.1.4) before we release the 2.2 (that might still take a few weeks).  

I pushed the cherry-picks to v2-1-test (just in case we need to release 2.1.5):

* Explain scheduler fine-tuning better (#18356) …
* Add PGBouncer recommendation in "setup-database' doc. (#18399) …
* Doc: Fix broken asc download link (#18350) …
* add sla_miss_callback section to the documentation (#18305) …

**Comment by potiuk** [comment]
Just +/- 18.000 lines this time.

**Comment by potiuk** [comment]
Looks good and we get some guidelines in hands of our users earlier :)

File: docs-archive/apache-airflow/2.1.4/_api/airflow/exceptions/index.html [hunk 1]
```diff
@@ -567,10 +567,10 @@
             </div>
         </div>
     </div>
-
+        
         <div class="d-flex">
-
-
+
```
Comment (kaxil) [inline docs-archive/apache-airflow/2.1.4/_api/airflow/exceptions/index.html:572]: Some space have been added? does not matter really but just saying :)

**Review by kaxil — COMMENTED** [review]
Most of the changes are whitepsace changes for some reason

File: docs-archive/apache-airflow/2.1.4/_api/airflow/exceptions/index.html (hunk 1 above)
Comment (potiuk) [reply docs-archive/apache-airflow/2.1.4/_api/airflow/exceptions/index.html:572]: No idea where from :(

Comment (potiuk) [reply docs-archive/apache-airflow/2.1.4/_api/airflow/exceptions/index.html:572]: Maybe a new sphinx version ?

Comment (potiuk) [reply docs-archive/apache-airflow/2.1.4/_api/airflow/exceptions/index.html:572]: There are "real" changes too:

https://github.com/apache/airflow-site/pull/481/files#diff-efbd11b756e4cd17f684fcb9773e97cc9fe67456920076a2633d57c216a3976aR815

**Review by kaxil — APPROVED** [review]
I am just going to approve it :-D -- too hard to review

**Pull request merged**